mods/ra/maps/situation-critical/situation-critical.lua

```diff
@@ -0,0 +1,223 @@
+	{ OuterPatrol5.Location, OuterPatrol4.Location, OuterPatrol3.Location, OuterPatrol2.Location, OuterPatrol1.Location, OuterPatrol7.Location, OuterPatrol6.Location },
+	{ OuterPatrol6.Location, OuterPatrol7.Location, OuterPatrol1.Location, OuterPatrol2.Location, OuterPatrol3.Location, OuterPatrol4.Location, OuterPatrol5.Location },
+	{ OuterPatrol3.Location, OuterPatrol4.Location, OuterPatrol5.Location, OuterPatrol6.Location, OuterPatrol7.Location, OuterPatrol1.Location, OuterPatrol2.Location },
+	{ OuterPatrol3.Location, OuterPatrol2.Location, OuterPatrol1.Location, OuterPatrol7.Location, OuterPatrol6.Location, OuterPatrol5.Location, OuterPatrol4.Location }
+}
+
+GroupPatrol = function(units, waypoints, delay)
+	local i = 1
+	local stop = false
+
+	Utils.Do(units, function(unit)
+		Trigger.OnIdle(unit, function()
+			if stop then
+				return
+			end
+			if unit.Location == waypoints[i] then
+				local bool = Utils.All(units, function(actor) return actor.IsIdle end)
+				if bool then
+					stop = true
+					i = i + 1
+					if i > #waypoints then
+						i = 1
+					end
+					Trigger.AfterDelay(delay, function() stop = false end)
+				end
+			else
+				unit.AttackMove(waypoints[i])
+			end
+		end)
+	end)
+end
+
+StartPatrols = function()
+	for i = 1, 5 do
+		GroupPatrol(OuterPatrols[i], OuterPatrolPaths[i], DateTime.Seconds(3))
+	end
+
+	for i = 1, 4 do
+		Trigger.AfterDelay(DateTime.Seconds(3* (i - 1)), function()
+			Trigger.OnIdle(Shocktroopers[i], function()
+				Shocktroopers[i].Patrol(InnerPatrolPaths[i])
+			end)
+		end)
+	end
+end
+
+LabInfiltrated = false
+SetupTriggers = function()
+	Trigger.OnAllKilled(SamSites, function()
+		local proxy = Actor.Create("powerproxy.parabombs", false, { Owner = ussr })
+		proxy.TargetAirstrike(TacticalNuke1.CenterPosition, Angle.SouthWest)
+		proxy.TargetAirstrike(TacticalNuke2.CenterPosition, Angle.SouthWest)
+		proxy.TargetAirstrike(TacticalNuke3.CenterPosition, Angle.SouthWest)
+		proxy.Destroy()
+	end)
+
+	Trigger.OnInfiltrated(BioLab, function()
+		Media.DisplayMessage("Plans stolen; erasing all data.", "Scientist")
+		Trigger.AfterDelay(DateTime.Seconds(5), function()
+			ussr.MarkCompletedObjective(InfiltrateLab)
+			LabInfiltrated = true
+		end)
+	end)
+
+	Trigger.OnAllKilled(PrimaryTargets, function()
+		ussr.MarkCompletedObjective(DestroyFacility)
+		ussr.MarkCompletedObjective(VolkovSurvive)
+	end)
+
+	Trigger.OnAllKilled(MissleSubs, function()
+		if not VolkovArrived == true then
+			ussr.MarkFailedObjective(KillPower)
+		end
+	end)
+end
+
+VolkovArrived = false
+SendInVolkov = function()
+	if not VolkovArrived == true then
+		ussr.MarkCompletedObjective(KillPower)
+		Media.PlaySpeechNotification(ussr, "ReinforcementsArrived")
+		teamVolkov = Reinforcements.ReinforceWithTransport(ussr, InsertionTransport, VolkovandFriend, VolkovEntryPath, { VolkovEntryPath[1] })[2][1]
+		VolkovArrived = true
+		Trigger.OnAddedToWorld(teamVolkov, function()
+			FindActors()
```

Comment:
Why do we have to find the actors if we already have them returned by `ReinforceWithTransport`?

Reply:
I need Volkov and the scientist tagged separately for their respective fail objective triggers.

Reply:
This can still be done by accessing the first or second entry in the table returned. (In fact, the local `teamVolkov` variable likely already is Volkov.)

Reply:
This is still not resolved(?).

Reply:
Nah I'm still trying to wrap my head around it.